foundation/auth/src/project.rs

```diff
@@ -89,18 +90,38 @@ pub async fn create_token_source_from_credentials(
     Ok(Box::new(ReuseTokenSource::new(ts, token)))
 }
 
+/// Creates token source using gke metadata server.
+/// 
+/// Will use the audience if provided, generating an identity token source.
+/// Otherwise, will use the scopes if provided, generating an access token source.
+/// 
+/// Returns an error if neither audience nor scopes are provided, or if any technical error occurs.
+pub async fn create_token_source_from_metadata_server(config: &Config<'_>,) -> Result<Box<dyn TokenSource>, error::Error> {
+    match config.audience {
```

Comment:
In google-cloud-pubsub and others, since the audience is set when the client is generated, this application will force the idtoken.
We checked google-cloud-go and did not see any use of idtoken, so we would like to position this as an option.

```rust
pub async fn create_token_source_from_project(
    project: &Project,
    config: Config<'_>,
) -> Result<Box<dyn TokenSource>, error::Error> {
    match project {
        Project::FromFile(file) => {
            if config.use_id_token {
                id_token_source_from_credentials(Default::default(), file, config.audience.unwrap_or_default()).await
            }else {
                create_token_source_from_credentials(file, &config).await
            }
        },
        Project::FromMetadataServer(_) => {
            let ts = if config.use_id_token {
                ComputeIdentitySource::new(&config.audience.unwrap_or_default())?
            }else {
                ComputeTokenSource::new(&config.scopes_to_string(","))?
            };
            let token = ts.token().await?;
            Ok(Box::new(ReuseTokenSource::new(Box::new(ts), token)))
        }
    }
}
```

How about the following code? This would allow the use of idtoken even when credentials are used.

Reply:
Sounds good to me.
I will update based on your feedback and test it by tomorrow.

Reply:
Tested. No issue found.

My tests covered both authentication tokens (OAuth and ID) from GKE, for in-house target services (cross project cloud run through PSC setup) as well as same project/cross project Google Cloud APIs (Vertex endpoints for example)